packages/app/features/splash/screen.tsx

```diff
@@ -259,7 +253,17 @@ function AuthButtons() {
       w="100%"
       alignSelf="center"
     >
-      <SignInButton setError={setSignInError} size="$4" w="$12" />
+      <SignInButton
+        setError={setSignInError}
+        size="$4"
+        w="$12"
+        isSigningIn={isSigningIn}
+        setIsSigningIn={setIsSigningIn}
+      >
```

Comment:
Not quite. Instead of passing the props let's just switch between two buttons
```suggestion
     isSigningIn ? 
     <SignInButton 
        size="$4"
        w="$12"
        disabled
        disabledStyle={..}>
      <SignInButton.Text>
        Signing In...
      </SignInButton.Text>
     </SignInButton> :
     <SignInButton
        size="$4"
        w="$12"
        onPress={handleSignIn}
      >...
```

Reply:
Not sure I get what you're saying here. The SignInButton is already responsible for all the sign in logic, the only thing that needs changing between it's two current uses is the text and button text styles

Reply:
The logic for signing in, like the mutation and and query param handlers shouldn’t be in the button. The button should just be styles and html. Let the parent component control all the logic.

Reply:
The button is currently used in multiple spots, so I can move the logic to a hook or whatever, but moving it to the parent means duplicating all that code in both spots

Reply:
It's okay to repeat code a little in this case. Buttons shouldn't be managing all that state. If you'd like to move the sign in logic into a hook as well, I think that's a smart idea. If we find ourselves rewriting the sign in stuff a lot we can always add the hook later

Reply:
Okay i'll copy pasta and then we can do a hook later if we need to